toy_2d/src/ilqr.py

```diff
@@ -156,33 +168,27 @@ def forward_pass(self, xx, uu, dd, KK, lr):
         utraj = [np.zeros((self.nu,))] * (self.N - 1)
         xtraj[0] = xx[0]
 
-        # TODO: compute forward pass
         for i in range(self.N-1):
             utraj[i] = uu[i] + KK[i]@(xtraj[i] - xx[i]) + lr*dd[i]
-            # if(utraj[i][0]<0 and utraj[i][1]>0): utraj[i] = np.array([0,0])
-            #TO DO
-            # xtraj[i+1] = quad_sim.F(xtraj[i], utraj[i], self.dt)
             xtraj[i+1] = self.sim_forward(xtraj[i], utraj[i])
         return xtraj, utraj
 
     def get_linearized_discrete_dynamics(self, state, action):
         q = np.array([state[0], state[2], state[4]])
         v = np.array([state[1], state[3], state[5]])
         self.lcs.set_linearization_point(q,v, action)
-        A, B_gen = self.lcs.get_A_matrix(), self.lcs.get_B_matrix()
-        pmat = self.lcs.get_P_matrix()
+        A, B_gen, C_mat, _, G_mat, H_mat, J_mat, _, pmat = self.lcs.get_lcs_terms()
         B = B_gen@pmat
-        C_mat = self.lcs.get_C_matrix()
-        G_mat = self.lcs.get_G_matrix()
-        J_mat = self.lcs.get_J_matrix()
-        H_mat = self.lcs.get_H_matrix()
         #access original dynamics to get the original lambdas
         curr_lam = self.get_lamda(state, action)
         eps = 1e-3
         # check if there is a contact at all. if not use the normal dynamics itself as cube is in air
-        if(np.argwhere(curr_lam>eps).shape[0]>0):
+        #the curr_lam vector is a vector of shape [16,] of which the first 8 are normal, the next 4 
+        #are tangential and the last 4 are slack variables. Hence to see if the cube made contact, we only
+        #check if the tangential or normal values are greater than zero. Slack variables might be non-zero
+        #even with no contact.
+        if(np.argwhere(curr_lam[:12]>eps).shape[0]>0):
```

Comment:
`curr_lam` is only size (16,) if the polytope has 4 corners.  In general, its shape is (4*n_corners).  It would be better for all of these hard-coded numbers to instead use the adaptive parameters, which are in `lcs.system.params.polytope.n_contacts`.